internal/client.go

```diff
@@ -626,6 +630,515 @@ type (
+	SchedulePolicies struct {
+		// Overlap - Controls what happens when an Action would be started by a Schedule at the same time that an older Action is still
+		// running.
+		Overlap enumspb.ScheduleOverlapPolicy
+
+		// CatchupWindow - The Temporal Server might be down or unavailable at the time when a Schedule should take an Action. When the Server
+		// comes back up, CatchupWindow controls which missed Actions should be taken at that point.
+		CatchupWindow time.Duration
+
+		// PauseOnFailure - When an Action times out or reaches the end of its Retry Policy.
+		PauseOnFailure bool
+	}
+
+	// ScheduleState describes the current state of a schedule.
+	ScheduleState struct {
+		// Note - Informative human-readable message with contextual notes, e.g. the reason
+		// a Schedule is paused. The system may overwrite this message on certain
+		// conditions, e.g. when pause-on-failure happens.
+		Note string
+
+		// Paused - True if the schedule is paused.
+		Paused bool
+
+		// LimitedActions - While true RemainingActions will be decremented for each action taken.
+		// Skipped actions (due to overlap policy) do not count against remaining actions.
+		LimitedActions bool
+
+		// RemainingActions - The Actions remaining in this Schedule. Once this number hits 0, no further Actions are taken.
+		// manual actions through backfill or ScheduleHandle.Trigger still run.
+		RemainingActions int64
+	}
+
+	// Schedule describes a created schedule.
+	Schedule struct {
+		// Action - Which Action to take
+		Action ScheduleAction
+
+		// Schedule - Describes when Actions should be taken.
+		Spec *ScheduleSpec
+
+		// SchedulePolicies - this schedules policies
+		Policy *SchedulePolicies
+
+		// State - this schedules state
+		State *ScheduleState
+	}
+
+	// ScheduleUpdate describes the desired new schedule from ScheduleHandle.Update.
+	ScheduleUpdate struct {
+		// Schedule - New schedule to replace the existing schedule with
+		Schedule *Schedule
+	}
+
+	// ScheduleUpdateOptions configure the parameters for updating a schedule.
+	ScheduleUpdateOptions struct{}
+
+	// ScheduleHandle represents a created schedule.
+	ScheduleHandle interface {
+		// GetID returns the schedule ID asssociated with this handle.
+		GetID() string
+
+		// Delete the Schedul
+		Delete(ctx context.Context) error
+
+		// Update the Schedule. fn takes a description of the schedule and returns the new desired schedule.
+		// If update returns a nil response then no update will occur.
+		//
+		// NOTE: If two Update calls are made in parallel to the same Schedule there is the potential
+		// for a race condition.
+		Update(ctx context.Context, fn func(*ScheduleDescribeResponse) *ScheduleUpdate, options ScheduleUpdateOptions) error
+
+		// Describe fetches the Schedule's description from the Server
+		Describe(ctx context.Context) (*ScheduleDescribeResponse, error)
+
+		// Trigger an Action to be taken immediately. Will override the schedules default policy
```

Comment:
note that the override is optional.. you can leave it UNSPECIFIED and it'll use the policy in the schedule